makederiv

```diff
@@ -499,6 +505,14 @@ while [[ "${@}" != "" ]] ; do
             MIDDLEOPTIONS+=(-b:a 448000)
             MIDDLEOPTIONS+=(-ar 48000)
             _get_audio_mapping -v "${SOURCEFILE}"
+        elif [[ "${OUTPUT_TYPE}" == "m4a" ]] ; then
```

Comment:
rather than 'm4a', let's make this a subcondition on if output_type = podcast, and then adjust the output whether the input is audio only or audio/video

Reply:
so that would mean moving these lines:

```
_get_videostreamcount "${SOURCEFILE}"
_get_audiostreamcount "${SOURCEFILE}"
```

before the first `if podcast`, then within the `if podcast` say `if [[ "${VIDEOSTREAMCOUNT}" -gt 0 ]] ; then` use the video extension and options, otherwise use the options for an audio only output. that subcondition would be needed in the conditional section that impact the container and as well in the section that impacts the audio options.

Reply:
Catriona and I tried to do this and successfully got -T podcast to create m4a files for audio, but it also now produces m4a files for video as well.  We suspect we are not using `_get_videostreamcount "${SOURCEFILE}"` and `if [[ "${VIDEOSTREAMCOUNT}" -gt 0 ]]` correctly... I tried placing `_get_videostreamcount "${SOURCEFILE}"` in many locations before the first `if podcast `code but none of it worked.  Looking forward to your thoughts.